multinode_integration_tests/src/mock_blockchain_client_server.rs

```diff
@@ -537,15 +568,47 @@ mod tests {
 
         let requests = subject.requests();
         assert_eq! (requests, vec! [
-            "POST /biddle HTTP/1.1\nContent-Type: application-json\nContent-Length: 82\n\n{\"jsonrpc\": \"2.0\", \"method\": \"first\", \"params\": [\"biddle\", \"de\", \"bee\"], \"id\": 40}".to_string(),
-            "POST /biddle HTTP/1.1\nContent-Type: application-json\nContent-Length: 48\n\n{\"jsonrpc\": \"2.0\", \"method\": \"second\", \"id\": 42}".to_string(),
-            "POST /biddle HTTP/1.1\nContent-Type: application-json\nContent-Length: 47\n\n{\"jsonrpc\": \"2.0\", \"method\": \"third\", \"id\": 42}".to_string(),
+            "POST /biddle HTTP/1.1\r\nContent-Type: application-json\r\nContent-Length: 82\r\n\r\n{\"jsonrpc\": \"2.0\", \"method\": \"first\", \"params\": [\"biddle\", \"de\", \"bee\"], \"id\": 40}".to_string(),
+            "POST /biddle HTTP/1.1\r\nContent-Type: application-json\r\nContent-Length: 48\r\n\r\n{\"jsonrpc\": \"2.0\", \"method\": \"second\", \"id\": 42}".to_string(),
+            "POST /biddle HTTP/1.1\r\nContent-Type: application-json\r\nContent-Length: 47\r\n\r\n{\"jsonrpc\": \"2.0\", \"method\": \"third\", \"id\": 42}".to_string(),
+        ])
+    }
+
+    #[test]
+    fn mbcs_understands_real_world_request() {
+        let _cluster = MASQNodeCluster::start();
+        let port = find_free_port();
+        let subject = MockBlockchainClientServer::builder(port)
+            .response(
+                Person {
+                    name: "Billy".to_string(),
+                    age: 15,
+                },
+                42,
+            )
+            .start();
+        let mut client = connect(port);
+
+        let request =
+            b"POST / HTTP/1.1\r\ncontent-type: application/json\r\nuser-agent: web3.rs\r\nhost: 172.18.0.1:32768\r\ncontent-length: 308\r\n\r\n{\"jsonrpc\":\"2.0\",\"method\":\"eth_getLogs\",\"params\":[{\"address\":\"0x59882e4a8f5d24643d4dda422922a870f1b3e664\",\"fromBlock\":\"0x3e8\",\"toBlock\":\"latest\",\"topics\":[\"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef\",null,\"0x00000000000000000000000027d9a2ac83b493f88ce9b4532edcf74e95b9788d\"]}],\"id\":0}";
+        client.write(request).unwrap();
```

Comment:
I know we kind of resigning on the strict G-unit format but in this test it took me more than normally to identify what could be the trigger. Well, wasn't so hard but you know, the expected formatting would make it look cleaner.

Reply:
Done.